elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryCancelThread.java

```diff
@@ -61,13 +66,13 @@ protected void cancelAsyncQuery() {
 
             TransactionRegistry transactionRegistry = elide.getTransactionRegistry();
             Map<UUID, DataStoreTransaction> runningTransactionMap = transactionRegistry.getRunningTransactions();
-
-            String filterDateFormatted = evaluateFormattedFilterDate(Calendar.SECOND, maxRunTimeSeconds);
-            String filterExpression = "status=in=(" + QueryStatus.CANCELLED.toString() + ","
-                    + QueryStatus.PROCESSING.toString() + ","
-                    + QueryStatus.QUEUED.toString() + ");createdOn=ge='" + filterDateFormatted + "'";
-
-            Collection<AsyncQuery> asyncQueryCollection = asyncQueryDao.getActiveAsyncQueryCollection(filterExpression);
+            String filterDateFormatted = evaluateFormattedFilterDate(Calendar.SECOND, 2 * maxRunTimeSeconds);
+            String filterExpression = "status=in=(" + QueryStatus.PROCESSING.toString() + ","
```

Comment:
Relying on time to find the queries that need to be canceled has some race conditions and other issues with configuration that can cause problems.

Instead of filtering on time, I think we need to introduce another state - CANCEL_COMPLETE or something along those lines that marks which queries have been successfully canceled.